BLSQ/iaso
# IA-3514: Allow to select editable org unit types on user roles

User should be able to assigne editable org unit types to a user roles, back-end is done.
Alos user in a user roles with editable org unit types should have the restriction from his profile AND user role editable org unit types

Related JIRA tickets : IA-3514

## Self proofreading checklist



## Changes

- adapt hook that checks that user has writable permission on a type
- added a field to select editable org unit types on user role dialog

## How to test

- Edit a user roles, check some types
- Connect with a user that have this user role
- Restriction should be the same as on this [PR](https://github.com/BLSQ/iaso/pull/1687)

## Print screen / video

<img width="1138" alt="Screenshot 2024-10-21 at 13 11 49" src="https://github.com/user-attachments/assets/0c708c79-5abc-465d-ab66-fdbd271b5e2b">
